Mathlib/Tactic/Abel.lean

```diff
@@ -473,7 +473,7 @@ open Elab.Tactic Parser.Tactic
 /-- Use `abel_nf` to rewrite the main goal. -/
 def abelNFTarget (s : IO.Ref AtomM.State) (cfg : AbelNF.Config) : TacticM Unit := withMainContext do
   let goal ← getMainGoal
-  let tgt ← instantiateMVars (← goal.getType)
+  let tgt ← instantiateMVars (← goal.getType')
```

Comment:
It's not clear to me why this helps, `abelNFCore` already does a `whnfR`.

Reply:
You can simplify the test:
```lean
example : True := by
  have := 0
  abel_nf
```

I do not know exactly what happens, but the `mdata` in the input gets passed on to the output of `abelNFCore` as `r`.  `r` is then filtered through ``if r.expr.isConstOf `True then`` and this sees either `mdata` or `True`, failing in the former case.

Reply:
Moved to #12084.